src/Illuminate/Filesystem/FilesystemAdapter.php

```diff
@@ -162,7 +162,9 @@ public function response($path, $name = null, array $headers = [], $disposition
 
         $response->setCallback(function () use ($path) {
             $stream = $this->readStream($path);
-            fpassthru($stream);
+            while (! feof($stream)) {
+                echo fread($stream, 2048);
```

Comment:
Do we know why this "dark magic hackery" fixes it 🧙‍♂️ ?

Reply:
@mfn I was researched now, fpassthru will write all contents to output buffer.

> Like

```php
file_put_contents('php://output', "All file data here");
```
In here, **fpassthru** will get all file data for sending to output buffer, so will temporary memory usage till send process all data into output buffer done.

We can optimize this with chunk via fread.
```php
while (! feof($stream)) {
    file_put_contents('php://output', fread($stream, 2048));
}
```

Reply:
Seems like output buffering tries chunking data with php free memory size.
if you will pass upper than free memory, output buffering will be failed.
This `while` giving data to output buffer with small sizes for help to output buffering chunks.

Reply:
Why not stream_copy_to_stream? Eg https://github.com/symfony/http-foundation/blob/0f977dd6a6ab22789f49cefccdb623734da2ba3c/BinaryFileResponse.php#L297-L303

Reply:
> Why not stream_copy_to_stream? Eg https://github.com/symfony/http-foundation/blob/0f977dd6a6ab22789f49cefccdb623734da2ba3c/BinaryFileResponse.php#L297-L303

Same things actually.